### PR TITLE
Working on-off interaction for desk lamp

### LINKED ISSUE #688: Add On/Off Interaction to table lamps
### Goal 

Add On/Off interaction to table lamps. Look at flashlights as an example of this function currently.

### PR DESCRIPTION
**Pull request by frimannloes**
### Summary

Adds on-off interaction to Desk Lamps.

Uses the IToggleable interface.

Gives Flashlight.cs the same treatment, which previously didn't have an RpcCall either.

## Pictures/Videos (optional)

https://user-images.githubusercontent.com/26187000/121584225-8ba53380-ca31-11eb-8f70-4584178f4932.mp4

## Fixes (optional)

Closes #688

### CODE REVIEW

**Comment by LightVelox** [comment]
missing a RPCCall, look at Boombox.cs for reference

**Comment by frimannloes** [comment]
Is my understanding of Rpc-functions correct? That they are to affirm states in a way?

File: Assets/Content/Items/Functional/Tools/Generic/desk lamp.prefab [hunk 1]
```diff
@@ -266,21 +266,24 @@ MonoBehaviour:
...
   Volume: 15
-  container: {fileID: 0}
   sprite: {fileID: 21300000, guid: 66026a73b54c5814aa62ed061b556bbd, type: 3}
   prefab: {fileID: 5021521585556056798}
   attachmentPoint: {fileID: 3367512477711822723}
+  attachmentPointAlt: {fileID: 0}
   bulkSize: 3
   traits: []
---- !u!114 &705127767441947080
+  Size: {x: 0, y: 0}
```
Comment (frimannloes) [inline Assets/Content/Items/Functional/Tools/Generic/desk lamp.prefab:283]: What do these sizes do? Should they be nonzero?

Comment (cosmiccoincidence) [reply Assets/Content/Items/Functional/Tools/Generic/desk lamp.prefab:283]: They are for the inventory system... we need to iron out a few details with the system still so many items have been given an appropriate size yet.

**Review by cosmiccoincidence — COMMENTED** [review]
I made a comment on the discord about a more generic 'toggle item' script that works in addition to the item script instead of replacing the item script with a flashlight script and needing to do that for anything that has a light (not to mention anything that has a toggle).

**Review by cosmiccoincidence — APPROVED** [review]
Good work!

**Pull request merged**